litani

```diff
@@ -281,9 +280,23 @@ def get_args():
         flags = arg.pop("flags")
         graph_pars.add_argument(*flags, **arg)
 
+    get_jobs_pars = subs.add_parser("get-jobs", help="print jobs")
```

Comment:
One request (specifically from @markrtuttle). Can both of these commands take a file? `--from-file` for `set-jobs` and `--out-file` for `get-jobs`. (I suggest the short flag is `-f` in both cases).

A very clean implementation could be:
* In the argument parser, set `default` to `sys.stdin` or `sys.stdout`
* `type` should be a function that takes a path to a file, opens it for reading/writing, and returns the handle. (So use `open` as a straight function call, without using `with`). For `set-jobs`, if the file to be read doesn't exist then raise.
* This means that the value of `args.from_file` will always be a handle (whether a real file or stdin) and you can just uniformly call `read` on it to read the JSON.

Reply:
(When I said 'raise', I specifically mean raise `argparse.ArgumentError`).